lake/api/api.go

```diff
@@ -19,7 +18,7 @@ import (
 )
 
 type Interface interface {
-	Query(ctx context.Context, d driver.Driver, head *lakeparse.Commitish, src string, srcfiles ...string) (zbuf.Progress, error)
+	Query(ctx context.Context, head *lakeparse.Commitish, ctrl bool, src string, srcfiles ...string) (zbuf.ProgressReader, error)
```

Comment:
Adding `ctrl` here feels wrong.
```suggestion
	Query(ctx context.Context, head *lakeparse.Commitish, ctrl bool, src string, srcfiles ...string) (zbuf.ProgressReader, error)
```

Reply:
Ok, see dc3bb75